buildSrc/subprojects/plugins/src/main/kotlin/org/gradle/gradlebuild/unittestandcompile/UnitTestAndCompilePlugin.kt

```diff
@@ -186,26 +187,62 @@ class UnitTestAndCompilePlugin : Plugin<Project> {
     }
 
     private
-    fun Project.configureTests() {
-        val javaInstallationForTest = rootProject.availableJavaInstallations.javaInstallationForTest
+    fun Test.getPreviousFailedTestClasses(): Set<String> {
+        val serializer = TestResultSerializer(getBinResultsDir())
+        if (serializer.isHasResults) {
```

Comment:
`return` can be lifted out for cleaner code:

`return if () { ... } else { ... }`

Reply:
If you want to write the method more compact you can also do this :)

`fun Test.getPreviousFailedTestClasses(): Set<String> = TestResultSerializer(binResultsDir).let { serializer ->` 

And then remove the return(s) completely.

Reply:
I realize `if (serializer.isHasResults)` is unnecessary so I removed it. Thanks for the suggestion, fixed.